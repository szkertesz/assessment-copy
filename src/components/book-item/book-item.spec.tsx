import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import BookItem from './book-item';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);
let store: any;

beforeEach(() => {
    store = mockStore({
        books: {
            books: [],
            status: 'succeeded',
            error: null,
            filterOptions: {
                status: 'reading',
            },
        },
    });
});
describe('BookItem', () => {
    it('should render component', () => {
        const { container } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <BookItem
                        bookInfo={{
                            id: 1,
                            title: "Harry Potter and the Philosopher's Stone",
                            author: 'J. K. Rowlingswsdfsdf',
                            description:
                                "Harry Potter and the Philosopher's Stone is a fantasy novel written by British author J. K. Rowling. The first novel in the Harry Potter series and Rowling's debut novel, it follows Harry Potter, a young wizard who discovers his magical heritage on his eleventh birthday,",
                            reading: true,
                            read: false,
                        }}
                    />
                </MemoryRouter>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
    it('should display loading message after status is changed', async () => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <BookItem
                        bookInfo={{
                            id: 1,
                            title: "Harry Potter and the Philosopher's Stone",
                            author: 'J. K. Rowlingswsdfsdf',
                            description:
                                "Harry Potter and the Philosopher's Stone is a fantasy novel written by British author J. K. Rowling. The first novel in the Harry Potter series and Rowling's debut novel, it follows Harry Potter, a young wizard who discovers his magical heritage on his eleventh birthday,",
                            reading: true,
                            read: false,
                        }}
                    />
                </MemoryRouter>
            </Provider>
        );
        expect(screen.getByText(/Harry Potter/)).toBeInTheDocument();
        fireEvent.click(screen.getByRole('radio', {name: 'Read'}));
        expect(await screen.findByText(/Loading.../)).toBeInTheDocument();
    });
});
