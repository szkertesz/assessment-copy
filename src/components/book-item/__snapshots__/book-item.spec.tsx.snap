// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem should render component 1`] = `
<div>
  <article
    class="book-item"
  >
    <h2
      class="book-item__title"
    >
      Harry Potter and the Philosopher's Stone
    </h2>
    <div
      class="book-item__actions"
    >
      <a
        class="book-item__button button"
        href="/edit/1"
      >
        Edit
      </a>
      <fieldset
        class="status-toggle"
      >
        <div
          class="status-toggle__group"
        >
          <input
            checked=""
            class="visually-hidden"
            id="1radio-reading"
            name="status-1"
            type="radio"
            value="reading"
          />
          <label
            class="status-toggle__label button"
            for="1radio-reading"
          >
            Reading
          </label>
        </div>
        <div
          class="status-toggle__group"
        >
          <input
            class="visually-hidden"
            id="1radio-read"
            name="status-1"
            type="radio"
            value="read"
          />
          <label
            class="status-toggle__label button"
            for="1radio-read"
          >
            Read
          </label>
        </div>
      </fieldset>
    </div>
  </article>
</div>
`;
